pkg/tfshim/sdk-v2/instance_state.go

```diff
@@ -36,68 +37,41 @@ func (s v2InstanceState) ID() string {
 }
 
 func (s v2InstanceState) Object(sch shim.SchemaMap) (map[string]interface{}, error) {
```

Comment:
This method wan't doing quite the right thing. The empty object value ended up parsed as `nil`. This caused problems because a `List[EmptyObj]` value under MaxItems=1 flattening got translated to `nil` which is indistinguishable from a `nil` list.

Reply:
The new implementation leans into TF primitives. I wonder how to ramp up testing to know we can make this replacement.

Reply:
The new implementation trips up some tests because it now generates `"foo": nil` entries in maps instead of eliding them.